onnx/defs/tensor/defs.cc

```diff
@@ -397,20 +397,25 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
-          auto axisAttr = ctx.getAttribute("axis");
-          int axis = axisAttr ? static_cast<int>(axisAttr->i()) : 0;
-          if (axis < 0) {
-            return;
-          }
           std::vector<int64_t> split;
           if (!getRepeatedAttribute(ctx, "split", split)) {
             if (!ctx.getInputType(0)->tensor_type().has_shape()) {
               return;
             }
             const auto& shape = ctx.getInputType(0)->tensor_type().shape();
-            if (axis >= shape.dim_size()) {
-              fail_type_inference("Invalid value of attribute 'axis'");
+            int r = shape.dim_size();
+            int axis = static_cast<int>(getAttribute(ctx, "axis", 0));
+            if (axis < -r || axis >= r) {
+              fail_type_inference(
+                  "Invalid value of attribute 'axis'. Rank=",
+                  r,
+                  " Value=",
+                  axis);
+            }
+            if (axis < 0) {
```

Comment:
Thank you very much for the change! Please also add more clarification for "axis" attribute in the op definition, so that ONNX users do not need to find the negative axis handling logic from shape inference codes.